prisma/devSetup.ts

```diff
@@ -0,0 +1,211 @@
+/* eslint-disable func-names */
+/* eslint-disable no-console */
+import { PrismaClient } from '@prisma/client';
+import loadCSVFromFile from './loadCsv';
+
+const seedHealthcareProfessionals = async function (prisma: PrismaClient, verbose = false) {
+  const firstEn = 0;
+  const middleEn = 1;
+  const lastEn = 2;
+  const firstJa = 3;
+  const middleJa = 4;
+  const lastJa = 5;
+  const degreeCol = 6;
+  const specialtyCol = 7;
+  const spokenLangCol = 8;
+
+  const devData:string[][] = loadCSVFromFile('./prisma/seedData/devHealthcareProfessionals.csv');
+
+  devData.forEach(async (row: string[], index: number) => {
+    // generate some unique ids using even/odd strategy
+    const enID = 2 * index;
+    const jaID = 2 * index + 1;
+
+    // Make a nested write
+    const newPersonName = await prisma.personName.upsert({
+      where: { id: index },
+      update: {},
+      create: {
+        id: index,
+        names: {
+          create: [
+            {
+              id: enID,
+              locale: 'en',
+              firstName: row[firstEn],
+              middleName: row[middleEn],
+              lastName: row[lastEn],
+            },
+            {
+              id: jaID,
+              locale: 'ja',
+              firstName: row[firstJa],
+              middleName: row[middleJa],
+              lastName: row[lastJa],
+            },
+          ],
+
+        },
+      },
+    });
+
+    if (verbose) {
+      console.log(`Inserted ${newPersonName.id} into PersonName`);
+    }
+
+    // create a healthcare professional
+    const newHealthPro = await prisma.healthcareProfessional.upsert({
+      where: { id: index },
+      update: {},
+      create: {
+        id: index,
+        personNameId: newPersonName.id,
+        isPublished: true,
+      },
+    });
+
+    // Link spoken languages
+    const spokenLangList = row[spokenLangCol].split(',');
+    spokenLangList.forEach(async (lang) => {
+      const dbSpokenLang = await prisma.spokenLanguage.findFirst(
+        {
+          where: {
+            iso639_3: lang,
+          },
+        },
+      );
+      if (dbSpokenLang) {
+        // TODO change to upsert
+        await prisma.healthcareProfessionalSpokenLanguage.create(
+          {
+            data: {
+              spokenLanguageIso639_3: dbSpokenLang.iso639_3,
+              healthcareProfessionalId: newHealthPro.id,
+            },
+          },
+        );
+      }
+    });
+
+    // Link Degrees
+    const degreeList = row[degreeCol].split(',');
+    degreeList.forEach(async (degree) => {
```

Comment:
bug: in node, `forEach(async` doesn't work (unless they recently fixed it.) it doesn't guarantee that every iteration is completed before the next. It launches them all essentially in parallel. 

Personally, I've always seen helper functions for each company or I've created one myself. Instead of `foreach(async` you can use `await Promise.all(degreeList.map(async` or a helper function if you want to make one

Reply:
^ I'll approve so we can keep going and if you want , we can fix this in an iterative PR

Reply:
woah! just learned that ES2018 there's now a `for await` simplified option! I'm excited to try this 🤩 ✨ 
https://stackoverflow.com/a/50874507/3417143

Reply:
I actually learned this while running this, with the counter defined outside the loop and saw there was a race condition, so I should update these to use the `index` inside the loop...